sdk/cloudmachine/Azure.CloudMachine/src/CloudMachineClient.cs

```diff
@@ -27,6 +29,11 @@ public partial class CloudMachineClient : ClientWorkspace
     [EditorBrowsable(EditorBrowsableState.Never)]
     public ConnectionCollection Connections { get; } = [];
 
+    private readonly ReaderWriterLockSlim _connectionsLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
```

Comment:
Can we refactor all the caching and locking code into some separate "cache" class?